[Automation] javac compile fails for org.codehaus.plexus:plexus-java:1.0.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26082529291/job/76688415990
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.codehaus.plexus:plexus-java:0.9.10
Raw versions JSON: [
  "1.0.0",
  "1.0.1",
  "1.0.2",
  "1.0.3",
  "1.0.4",
  "1.0.5",
  "1.0.6",
  "1.0.7",
  "1.1.0",
  "1.1.1",
  "1.1.2",
  "1.2.0",
  "1.3.0",
  "1.4.0",
  "1.5.0",
  "1.5.1",
  "1.5.2"
]
Parsed version count: 17
Parsed versions: 1.0.0 1.0.1 1.0.2 1.0.3 1.0.4 1.0.5 1.0.6 1.0.7 1.1.0 1.1.1 1.1.2 1.2.0 1.3.0 1.4.0 1.5.0 1.5.1 1.5.2
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.codehaus.plexus:plexus-java:0.9.10:1.0.0
========================================================================================
========================================================================================
 org.codehaus.plexus:plexus-java:0.9.10:1.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 1.0.0 at 2026-05-19T07:29:16Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean test -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.codehaus.plexus:plexus-java:0.9.10
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                        ^
  symbol:   method getModuleName(Path)
  location: class MainClassModuleNameExtractor
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
          String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                          ^
    symbol:   method getModuleName(Path)
    location: class MainClassModuleNameExtractor
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)

> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.codehaus.plexus:plexus-java:0.9.10 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.0.0 at 2026-05-19T07:29:30Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 1.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean test -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"]
========================================================================================
 org.codehaus.plexus:plexus-java:0.9.10:1.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.0.0 at 2026-05-19T07:29:30Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 908ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                        ^
  symbol:   method getModuleName(Path)
  location: class MainClassModuleNameExtractor
1 error
> Task :compileTestJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
          String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                          ^
    symbol:   method getModuleName(Path)
    location: class MainClassModuleNameExtractor
  1 error

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.0.0 at 2026-05-19T07:29:33Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 1.0.0 failed with exit code 1.
FAILED[javac compile][1.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26082529291/job/76688415974
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.codehaus.plexus:plexus-java:0.9.10
Raw versions JSON: [
  "1.0.0",
  "1.0.1",
  "1.0.2",
  "1.0.3",
  "1.0.4",
  "1.0.5",
  "1.0.6",
  "1.0.7",
  "1.1.0",
  "1.1.1",
  "1.1.2",
  "1.2.0",
  "1.3.0",
  "1.4.0",
  "1.5.0",
  "1.5.1",
  "1.5.2"
]
Parsed version count: 17
Parsed versions: 1.0.0 1.0.1 1.0.2 1.0.3 1.0.4 1.0.5 1.0.6 1.0.7 1.1.0 1.1.1 1.1.2 1.2.0 1.3.0 1.4.0 1.5.0 1.5.1 1.5.2
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.codehaus.plexus:plexus-java:0.9.10:1.0.0
========================================================================================
========================================================================================
 org.codehaus.plexus:plexus-java:0.9.10:1.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 1.0.0 at 2026-05-19T07:29:12Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean test -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.codehaus.plexus:plexus-java:0.9.10
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                        ^
  symbol:   method getModuleName(Path)
  location: class MainClassModuleNameExtractor
1 error
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
          String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                          ^
    symbol:   method getModuleName(Path)
    location: class MainClassModuleNameExtractor
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.codehaus.plexus:plexus-java:0.9.10 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.0.0 at 2026-05-19T07:29:26Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 1.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean test -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"]
========================================================================================
 org.codehaus.plexus:plexus-java:0.9.10:1.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.0.0 at 2026-05-19T07:29:26Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 978ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                        ^
  symbol:   method getModuleName(Path)
  location: class MainClassModuleNameExtractor
1 error

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
          String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                          ^
    symbol:   method getModuleName(Path)
    location: class MainClassModuleNameExtractor
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.0.0 at 2026-05-19T07:29:29Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 1.0.0 failed with exit code 1.
FAILED[javac compile][1.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26082529291/job/76688416019
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.codehaus.plexus:plexus-java:0.9.10
Raw versions JSON: [
  "1.0.0",
  "1.0.1",
  "1.0.2",
  "1.0.3",
  "1.0.4",
  "1.0.5",
  "1.0.6",
  "1.0.7",
  "1.1.0",
  "1.1.1",
  "1.1.2",
  "1.2.0",
  "1.3.0",
  "1.4.0",
  "1.5.0",
  "1.5.1",
  "1.5.2"
]
Parsed version count: 17
Parsed versions: 1.0.0 1.0.1 1.0.2 1.0.3 1.0.4 1.0.5 1.0.6 1.0.7 1.1.0 1.1.1 1.1.2 1.2.0 1.3.0 1.4.0 1.5.0 1.5.1 1.5.2
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.codehaus.plexus:plexus-java:0.9.10:1.0.0
========================================================================================
========================================================================================
 org.codehaus.plexus:plexus-java:0.9.10:1.0.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 1.0.0 at 2026-05-19T07:29:19Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.0.0" ./gradlew clean test -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.codehaus.plexus:plexus-java:0.9.10
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                        ^
  symbol:   method getModuleName(Path)
  location: class MainClassModuleNameExtractor
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date
          String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                          ^
    symbol:   method getModuleName(Path)
    location: class MainClassModuleNameExtractor
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.codehaus.plexus:plexus-java:0.9.10 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 12s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.0.0 at 2026-05-19T07:29:32Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 1.0.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.0.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.0.0" ./gradlew clean test -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"]
========================================================================================
 org.codehaus.plexus:plexus-java:0.9.10:1.0.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.0.0 at 2026-05-19T07:29:32Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 892ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                        ^
  symbol:   method getModuleName(Path)
  location: class MainClassModuleNameExtractor
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java:35: error: cannot find symbol
          String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
                                                          ^
    symbol:   method getModuleName(Path)
    location: class MainClassModuleNameExtractor
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.codehaus.plexus/plexus-java/0.9.10/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.0.0 at 2026-05-19T07:29:35Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 1.0.0 failed with exit code 1.
FAILED[javac compile][1.0.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.0.0" ./gradlew clean compileTestJava -Pcoordinates="org.codehaus.plexus:plexus-java:0.9.10"]
```

--- a/metadata/org.codehaus.plexus/plexus-java/1.0.0/reachability-metadata.json
+++ b/metadata/org.codehaus.plexus/plexus-java/1.0.0/reachability-metadata.json
@@ -1,0 +1,54 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor"
+      },
+      "type": "java.lang.module.ModuleDescriptor",
+      "methods": [
+        {
+          "name": "name",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor"
+      },
+      "type": "java.lang.module.ModuleReference",
+      "methods": [
+        {
+          "name": "descriptor",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor"
+      },
+      "type": "java.lang.reflect.Method[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor"
+      },
+      "type": "jdk.internal.module.ModuleReferenceImpl"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor"
+      },
+      "glob": "META-INF/versions/9/org/codehaus/plexus/languages/java/jpms/CmdModuleNameExtractor.class"
+    }
+  ]
+}

--- a/metadata/org.codehaus.plexus/plexus-java/index.json
+++ b/metadata/org.codehaus.plexus/plexus-java/index.json
@@ -15,6 +15,15 @@
     ]
   },
   {
+    "metadata-version" : "1.0.0",
+    "tested-versions" : [
+      "1.0.0"
+    ],
+    "allowed-packages" : [
+      "org.codehaus.plexus.languages.java"
+    ]
+  },
+  {
     "metadata-version" : "0.9.11",
     "source-code-url" : "https://repo1.maven.org/maven2/org/codehaus/plexus/plexus-java/$version$/plexus-java-$version$-sources.jar",
     "repository-url" : "https://github.com/codehaus-plexus/plexus-languages",

--- a/metadata/org.codehaus.plexus/plexus-java/index.json
+++ b/metadata/org.codehaus.plexus/plexus-java/index.json
@@ -16,6 +16,11 @@
   },
   {
     "metadata-version" : "1.0.0",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/codehaus/plexus/plexus-java/$version$/plexus-java-$version$-sources.jar",
+    "repository-url" : "https://github.com/codehaus-plexus/plexus-languages",
+    "test-code-url" : "https://github.com/codehaus-plexus/plexus-languages/tree/plexus-languages-$version$/plexus-java/src/test",
+    "documentation-url" : "https://repo1.maven.org/maven2/org/codehaus/plexus/plexus-java/$version$/plexus-java-$version$-javadoc.jar",
+    "description" : "Plexus Java is the Java-language module of Plexus Languages, providing JVM utilities for reading JPMS module descriptors and deriving module names from Java artifacts. It also includes Java version parsing helpers used by Maven and Plexus tooling when inspecting Java libraries.",
     "tested-versions" : [
       "1.0.0"
     ],

--- a/stats/org.codehaus.plexus/plexus-java/1.0.0/execution-metrics.json
+++ b/stats/org.codehaus.plexus/plexus-java/1.0.0/execution-metrics.json
@@ -1,0 +1,104 @@
+{
+  "fix_javac_fail:2026-05-19": {
+    "timestamp": "2026-05-19T11:43:04.580255Z",
+    "library": "org.codehaus.plexus:plexus-java:1.0.0",
+    "starting_commit": "b4e62aac9c05784b3041640e94656ed74d6acc55",
+    "ending_commit": "e54fa46663ecb1d73c43d611c2910371e22bd792",
+    "previous_library": "org.codehaus.plexus:plexus-java:0.9.10",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "1.0.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 199,
+          "missed": 2278,
+          "ratio": 0.080339,
+          "total": 2477
+        },
+        "line": {
+          "covered": 41,
+          "missed": 519,
+          "ratio": 0.073214,
+          "total": 560
+        },
+        "method": {
+          "covered": 5,
+          "missed": 138,
+          "ratio": 0.034965,
+          "total": 143
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "0.9.10",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 6,
+            "totalCalls": 6
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 7,
+        "totalCalls": 7
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 291,
+          "missed": 1534,
+          "ratio": 0.159452,
+          "total": 1825
+        },
+        "line": {
+          "covered": 55,
+          "missed": 371,
+          "ratio": 0.129108,
+          "total": 426
+        },
+        "method": {
+          "covered": 6,
+          "missed": 91,
+          "ratio": 0.061856,
+          "total": 97
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 26102,
+      "cached_input_tokens_used": 124928,
+      "output_tokens_used": 1982,
+      "iterations": 1,
+      "cost_usd": 0.122,
+      "tested_library_loc": 41,
+      "metadata_entries": 8,
+      "previous_library_metadata_entries": 8,
+      "code_coverage_percent": 7.32,
+      "previous_library_coverage_percent": 12.91
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.codehaus.plexus/plexus-java/1.0.0/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java",
+      "metadata_file": "metadata/org.codehaus.plexus/plexus-java/1.0.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.codehaus.plexus/plexus-java/1.0.0/stats.json
+++ b/stats/org.codehaus.plexus/plexus-java/1.0.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "1.0.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 199,
+        "missed" : 2278,
+        "ratio" : 0.080339,
+        "total" : 2477
+      },
+      "line" : {
+        "covered" : 41,
+        "missed" : 519,
+        "ratio" : 0.073214,
+        "total" : 560
+      },
+      "method" : {
+        "covered" : 5,
+        "missed" : 138,
+        "ratio" : 0.034965,
+        "total" : 143
+      }
+    }
+  } ]
+}

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/.gitignore
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/build.gradle
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/build.gradle
@@ -1,0 +1,31 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.codehaus.plexus:plexus-java:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+tasks.named("nativeTest") {
+    runtimeArgs.add("-Djava.home=${System.getenv('JAVA_HOME')}")
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/gradle.properties
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.codehaus.plexus:plexus-java:1.0.0
+library.version = 1.0.0
+metadata.dir = org.codehaus.plexus/plexus-java/1.0.0/

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/settings.gradle
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.codehaus.plexus.plexus-java_tests'

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java
@@ -1,0 +1,67 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_codehaus_plexus.plexus_java;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.jar.Attributes;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.util.jar.Manifest;
+import org.codehaus.plexus.languages.java.jpms.MainClassModuleNameExtractor;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class MainClassModuleNameExtractorTest {
+    private static final String MODULE_NAME = "org.example.plexus.fixture";
+
+    @TempDir
+    Path temporaryDirectory;
+
+    @Test
+    void getsAutomaticModuleNameFromJarManifest() throws Exception {
+        Path moduleJar = createAutomaticModuleJar();
+
+        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
+
+        assertThat(moduleName).isEqualTo(MODULE_NAME);
+    }
+
+    @Test
+    void extractsModuleNamesThroughForkedMainClass() throws Exception {
+        Path moduleJar = createAutomaticModuleJar();
+        MainClassModuleNameExtractor extractor =
+                new MainClassModuleNameExtractor(Path.of(System.getProperty("java.home")));
+
+        Map<String, String> moduleNames = extractor.extract(Map.of("fixture", moduleJar));
+
+        assertThat(moduleNames).containsEntry("fixture", MODULE_NAME);
+    }
+
+    private Path createAutomaticModuleJar() throws IOException {
+        Path moduleJar = Files.createTempFile(temporaryDirectory, "automatic-module", ".jar");
+        Manifest manifest = new Manifest();
+        Attributes attributes = manifest.getMainAttributes();
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        attributes.putValue("Automatic-Module-Name", MODULE_NAME);
+
+        try (OutputStream outputStream = Files.newOutputStream(moduleJar);
+                JarOutputStream jarOutputStream = new JarOutputStream(outputStream, manifest)) {
+            JarEntry resourceEntry = new JarEntry("org/example/plexus/fixture/resource.txt");
+            jarOutputStream.putNextEntry(resourceEntry);
+            jarOutputStream.write("fixture".getBytes(StandardCharsets.UTF_8));
+            jarOutputStream.closeEntry();
+        }
+        return moduleJar;
+    }
+}

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/src/test/java/org_codehaus_plexus/plexus_java/MainClassModuleNameExtractorTest.java
@@ -31,10 +31,12 @@ public class MainClassModuleNameExtractorTest {
     @Test
     void getsAutomaticModuleNameFromJarManifest() throws Exception {
         Path moduleJar = createAutomaticModuleJar();
+        MainClassModuleNameExtractor extractor =
+                new MainClassModuleNameExtractor(Path.of(System.getProperty("java.home")));
 
-        String moduleName = MainClassModuleNameExtractor.getModuleName(moduleJar);
+        Map<Path, String> moduleNames = extractor.extract(Map.of(moduleJar, moduleJar));
 
-        assertThat(moduleName).isEqualTo(MODULE_NAME);
+        assertThat(moduleNames).containsEntry(moduleJar, MODULE_NAME);
     }
 
     @Test

--- a/tests/src/org.codehaus.plexus/plexus-java/1.0.0/user-code-filter.json
+++ b/tests/src/org.codehaus.plexus/plexus-java/1.0.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.codehaus.plexus.languages.java.**"
+    },
+    {
+      "includeClasses" : "org_codehaus_plexus.plexus_java.**"
+    }
+  ]
+}

